portfolio/src/main/java/com/google/sps/data/TimeRangeGroup.java

```diff
@@ -0,0 +1,28 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+
+public interface TimeRangeGroup {
+
+  /**
+   * Merges the collection of time ranges that is passed into the constructor so that all time
+   * ranges are disjoint.
+   */
+  public Collection<TimeRange> mergeInputTimeRanges(Collection<TimeRange> timeRanges);
```

Comment:
Consider renaming to addTimeRanges. I think the merging should be done internally, rather than explicitly specified by the callers.

Reply:
Updated. There will not be merging logic in this interface, since all time ranges passed in are free time ranges calculated in the CalendarEventsGroup class, and these free time ranges will be disjoint.